lib/IRGen/GenFunc.cpp

```diff
@@ -1436,7 +1436,12 @@ void irgen::emitFunctionPartialApplication(
     llvm::Value *ctx = args.claimNext();
     if (isIndirectFormalParameter(*singleRefcountedConvention))
       ctx = IGF.Builder.CreateLoad(ctx, IGF.IGM.getPointerAlignment());
-    ctx = IGF.Builder.CreateBitCast(ctx, IGF.IGM.RefCountedPtrTy);
+    // We might get a struct containing a pointer e.g type <{ %AClass* }>
+    if (!ctx->getType()->isPointerTy())
+      ctx = IGF.coerceValue(ctx, IGF.IGM.RefCountedPtrTy,
```

Comment:
does coerceValue() not handle the bitcast case as well?

Reply:
Yes it does: https://github.com/apple/swift/pull/21207